app/uk/gov/hmrc/merchandiseinbaggage/controllers/CheckYourAnswersAmendHandler.scala

```diff
@@ -92,6 +96,7 @@ class CheckYourAnswersAmendHandler @Inject()(
         _ <- calculationService.amendDeclaration(updatedDeclaration)
         redirectUrl <- paymentService
                         .sendPaymentRequest(updatedDeclaration.mibReference, Some(updatedAmendment.reference), calculationResults.results)
+        _ <- auditDeclaration(updatedDeclaration)
```

Comment:
this should be before triggering payment request, we need to place this line between 96 and 97
Also i guess `sendPaymentRequest` redirects straightaway to /declaration-confirmation in-case of £0 payments, but in that case we don't want this audit event because wont make any payment attempt

Reply:
moved